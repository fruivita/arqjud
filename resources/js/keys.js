/**
 * Chaves para provide/inject e outras constantes globais
 *
 * @see https://vuejs.org/guide/components/provide-inject.html#working-with-symbol-keys
 */

export const perPageKey = Symbol();
export const updatePerPageKey = Symbol();
export const exibirModalKey = Symbol();
export const fecharModalKey = Symbol();
export const mascaraCNJ = '#######-##.####.#.##.####';
