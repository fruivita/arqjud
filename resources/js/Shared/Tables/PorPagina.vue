<!--
    Seletor de paginação dos resultados.

    @link https://vuejs.org/guide/introduction.html
    @link https://tailwindcss.com/docs
    @link https://inertiajs.com/
    @link https://www.thisdot.co/blog/provide-inject-api-with-vue-3
 -->

<script setup>
import { perPageKey, updatePerPageKey } from '@/keys.js';
import { useDadosEstaticosStore } from '@/Stores/DadosEstaticosStore';
import { useTranslationsStore } from '@/Stores/TranslationsStore';
import { computed, inject } from 'vue';

const __ = useTranslationsStore().__;
const opcoes = useDadosEstaticosStore().paginacao;

const perPage = inject(perPageKey);
const updatePerPage = inject(updatePerPageKey);

const opcaoEscolhida = computed({
    get: () => perPage.value,
    set: updatePerPage,
});
</script>

<template>
    <div class="space-x-3 text-right">
        <label for="por_pagina">{{ __('Paginação') }}</label>

        <select
            id="por_pagina"
            v-model="opcaoEscolhida"
            class="rounded bg-primaria-300 text-right dark:bg-secundaria-500"
        >
            <option v-for="opcao in opcoes" :key="opcao" :value="opcao">
                {{ opcao }}
            </option>
        </select>
    </div>
</template>
