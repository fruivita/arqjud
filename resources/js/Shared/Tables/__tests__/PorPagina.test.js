/**
 * Testes para o componente PorPagina.
 *
 * @link https://vuejs.org/guide/scaling-up/testing.html
 * @link https://test-utils.vuejs.org/guide/
 * @link https://vitest.dev/
 */

import { perPageKey, updatePerPageKey } from '@/keys';
import PorPagina from '@/Shared/Tables/PorPagina.vue';
import { createTestingPinia } from '@pinia/testing';
import { mount } from '@vue/test-utils';
import { beforeEach, describe, expect, test, vi } from 'vitest';

vi.stubGlobal('_dados', { paginacao: [10, 20, 30] });

vi.mock('@inertiajs/inertia-vue3', () => ({
    usePage: () => ({
        props: {
            value: { per_page: 20 },
        },
    }),
}));

let mountFunction;

beforeEach(() => {
    const pinia = () => createTestingPinia({ stubActions: false, createSpy: vi.fn });
    mountFunction = (options = {}) => {
        return mount(PorPagina, {
            global: {
                plugins: [pinia()],
                provide: {
                    [perPageKey]: 25,
                    [updatePerPageKey]: vi.fn,
                },
            },
            ...options,
        });
    };
});

describe('HeadingOrdenavel', () => {
    test('renderiza o componente respeitando o snapshot', () => {
        expect(mountFunction().html()).toMatchSnapshot();
    });
});
