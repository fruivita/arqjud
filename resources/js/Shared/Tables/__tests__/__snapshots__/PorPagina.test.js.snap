// Vitest Snapshot v1

exports[`HeadingOrdenavel > renderiza o componente respeitando o snapshot 1`] = `
"<div class=\\"space-x-3 text-right\\"><label for=\\"por_pagina\\">Paginação</label><select id=\\"por_pagina\\" class=\\"rounded bg-primaria-300 text-right dark:bg-secundaria-500\\">
    <option value=\\"10\\">10</option>
    <option value=\\"20\\">20</option>
    <option value=\\"30\\">30</option>
  </select></div>"
`;
